rfcs/0137-nix-language-version.md

```diff
@@ -0,0 +1,92 @@
+---
+feature: nix-language-version
+start-date: 2022-12-12
+author: @fricklerhandwerk
+co-authors: @thufschmitt @Ericson2314
+shepherd-team: 
+shepherd-leader: 
+related-issues: https://github.com/NixOS/nix/issues/7255
+---
+
+# Summary
+[summary]: #summary
+
+Introduce a convention to determine which version of the Nix language grammar to use for parsing a Nix file.
+
+# Motivation
+[motivation]: #motivation
+
+Currently it's impossible to introduce backwards-incompatible changes to the Nix language without breaking existing setups.
+This proposal is first step towards overcoming that limitation.
+
+# Detailed design
+[design]: #detailed-design
+
+Introduce a magic comment in the first line of a Nix file:
```

Comment:
I think we had some kind of tooling for nix-scripts where first line is shebang?

Reply:
Please add a reference. I don't know of such a thing and wouldn't know what to search for.

Reply:
I see. That's indeed cursed. Does that make an argument for the RFC or can we close this thread, since it does not preclude the usage you envision?

Reply:
Agreed. Addressed here: https://github.com/NixOS/rfcs/pull/137#discussion_r1050809650